### DISCUSSION TITLE
Any reason why the runtime may spawn more threads than what I pass as limit in main macro?

### DISCUSSION THREAD

**Discussion by DevasiaThomas** [tokio-rs/tokio#4208] [Q&A] [answered]
```
#[tokio::main(flavor = "multi_thread", worker_threads = 4)]
async fn main() -> Result<()> {
```
That is how my main function is. But I see more than that at times(upto 8). Would there be any reason for tokio to use more threads than limited?

Thanks in advance :)

PS: I am not doing any spawn_blocking.

**Comment by Darksonn** [comment, marked as answer]
Tokio does not spawn any extra threads if you are not using `spawn_blocking`. Perhaps you are using it indirectly via `tokio::fs` or an external crate?

Note that the four threads are in addition to the main thread, so you can end up with five threads in your example, but not eight.

**Reply by DevasiaThomas** [reply]
Thanks for letting me know. I'll look further
